src/contradictory_claims/claims/models.py

```diff
@@ -0,0 +1,129 @@
+
+
+from typing import Dict, Optional
+
+
+import torch
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import ConditionalRandomField, Seq2VecEncoder, TimeDistributed
+from allennlp.modules.text_field_embedders import TextFieldEmbedder  # noqa:F401
+from allennlp.nn import InitializerApplicator, RegularizerApplicator  # noqa:F401
+from allennlp.nn import util
+from allennlp.training.metrics import CategoricalAccuracy
+from overrides import overrides
+from torch.nn.modules.linear import Linear
+
+
+@Model.register("discourse_crf_classifier")
+class DiscourseCrfClassifier(Model):
+    """Discourse CRF model class."""
+
+    global InitializerApplicator
+
+    def __init__(self,
+                 vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,  # noqaF821
+                 sentence_encoder: Seq2VecEncoder,
+                 initializer: InitializerApplicator = InitializerApplicator(), # noqaB008
+                 dropout: Optional[float] = None,
+                 regularizer: Optional[RegularizerApplicator] = None) -> None:
+        """
+        Initialize the class.
+
+        :param vocab: The vocabulary file
+        :param text_field_embedder: TextFieldEmbedder class object
+        :param: sentence_encoder: Seq2VecEncoder class object
+        :param initializer: InitializerApplicator object
+        :param dropout: float value of dropout
+        :param regularizer: RegularizerApplication
+        """
+        super(DiscourseCrfClassifier, self).__init__(vocab, regularizer)
+
+        self.text_field_embedder = text_field_embedder
+        self.num_classes = self.vocab.get_vocab_size("labels")
+        self.sentence_encoder = sentence_encoder
+        if dropout:
+            self.dropout = torch.nn.Dropout(dropout)
+        else:
+            self.dropout = None
+        self.metrics = {
+            "accuracy": CategoricalAccuracy(),
+            "accuracy3": CategoricalAccuracy(top_k=3),
+        }
+        self.loss = torch.nn.CrossEntropyLoss()
+        self.label_projection_layer = TimeDistributed(Linear(self.sentence_encoder.get_output_dim(),
+                                                             self.num_classes))
+
+        constraints = None  # allowed_transitions(label_encoding, labels)
+        self.crf = ConditionalRandomField(
+            self.num_classes, constraints,
+            include_start_end_transitions=False,
+        )
+        initializer(self)
+
+    @overrides
+    def forward(self,
+                sentences: Dict[str, torch.LongTensor],
+                labels: torch.LongTensor = None) -> Dict[str, torch.Tensor]:
+        """Forward pass of the model."""
+        # print(sentences['tokens'].size())
+        # print(labels.size())
+
+        embedded_sentences = self.text_field_embedder(sentences)
+        token_masks = util.get_text_field_mask(sentences, 1)
+        sentence_masks = util.get_text_field_mask(sentences)
```

Comment:
Could you add a bit of documentation about the intuition for what's going on in the model forward pass? What's being masked/embedded, etc. Or if you could point to a reference that would be good too.